servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpResponse.java

```diff
@@ -37,7 +37,7 @@ final class DefaultHttpResponse extends AbstractDelegatingHttpResponse
                         @Nullable final HttpHeaders trailers) {
         super(original);
         this.payloadBody = payloadBody;
-        this.trailers = trailers;
+        this.trailers = (trailers != null && trailers.isEmpty()) ? null : trailers;
```

Comment:
This treats and empty set of trailers as the same as `null`. By doing this on construction we are able to let `toStreamingResponse()` and `trailers()` behave as normal.

Reply:
We can not drop pre-allocated trailers object on every aggregation bcz users can do aggregation in place inside their controller and still use old reference to `HttpHeaders` to modify trailers state

Reply:
ugh, I was afraid of this...